code/scripts/DirectoryController.hs

```diff
@@ -0,0 +1,112 @@
+module DirectoryController (createFolder, createFile, finder, getDirectories, 
+  stripPath, DrasilPack, FileName, FolderName, File(..), Folder(..)) where
+
+import Data.List
+import System.IO
+import System.Directory
+
+type FilterPrefix = String
+type DrasilPack = String
+type FileName = FilePath
+type FolderName = FilePath
+
+-- File and Folder data types for storing drasil- package, name and filepath
+data File = File { fileDrasilPack :: DrasilPack
+                 , fileName :: FileName
+                 , filePath :: FilePath
+                 } deriving (Show)
+
+data Folder = Folder { folderDrasilPack :: DrasilPack
+                     , folderName :: FolderName
+                     , folderPath :: FilePath
+                     } deriving (Show)
+
+-- used to create new folders of folder data type
+createFolder :: FilePath -> DrasilPack -> FolderName -> Folder
+createFolder dp drpk fn = Folder {folderDrasilPack=drpk,folderName=fn,folderPath=dp}
+
+-- used to create new files of file data type
+createFile :: FilePath -> DrasilPack -> FileName -> File
+createFile fp drpk fn = File {fileDrasilPack=drpk,fileName=fn,filePath=fp}
+
+-- iterates through drasil- package; outputs subdirectories and haskell files
+iterator :: Folder -> IO ([Folder],[File])
+iterator folder = do
+  setCurrentDirectory (folderPath folder)
+  rawContents <- listDirectory (folderName folder)
+
+  let bakedContents = (folders,files)
+      folders = map (createFolder workingDirectory currentDrasilPack) folderNames
+      files = map (createFile workingDirectory currentDrasilPack) fileNames
+
+      folderNames = sort $ rawContents \\ filter (isInfixOf ".") rawContents
+      fileNames = sort $ filter (isSuffixOf ".hs") rawContents
+
+      workingDirectory = getFolderPath folder
+      currentDrasilPack = folderDrasilPack folder
+  return bakedContents
+
+-- recursively searches for all folders + files in a directory using iterator
+finder :: Folder -> IO [File]
+finder folder = do
+  rawData <- iterator folder
+
+  let rawFolders = fst rawData
+  folders <- verifyDirectories rawFolders
+  rawFiles <- mapM finder folders
+
+  let bakedFiles
```

Comment:
`let bakedFiles = concat rawFiles` is equivalent, I believe.

Reply:
I've changed the definition for `bakedFiles` to that [here](https://github.com/JacquesCarette/Drasil/pull/2208/commits/4065330bf5885d36437fb97d322a5c23d339df61#) (old lines 58-60; new line 59).